Homework/test_form.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service as ChromeService
+from webdriver_manager.chrome import ChromeDriverManager
+from text_box_page import TextBoxPage
+
+@pytest.fixture
+def setup():
+    service = ChromeService(executable_path=ChromeDriverManager().install())
+    driver = webdriver.Chrome(service=service)
+    driver.get("https://demoqa.com/text-box")
+    driver.maximize_window()
+    yield driver
+    driver.quit()
+
+
+def test_submit_form(setup):
+    text_box_page = TextBoxPage(setup)
+
+    # Input Data
+    full_name = "Donald Duck"
+    email = "donald.duck@example.com"
+    current_address = "56 Main St"
+    permanent_address = "379 Apple Rd"
+
+    # Actions
+    text_box_page.enter_full_name(full_name)
+    text_box_page.enter_email(email)
+    text_box_page.enter_current_address(current_address)
+    text_box_page.enter_permanent_address(permanent_address)
+    text_box_page.click_submit()
+
+    # Expected results
+    expected_name = f"Name:{full_name}"
+    expected_email = f"Email:{email}"
+    expected_current_address = f"Current Address :{current_address}"
+    expected_permanent_address = f"Permananet Address :{permanent_address}"
+
+    # Outputs
+    actual_name, actual_email, actual_current_address, actual_permanent_address = text_box_page.get_output()
```

Comment:
Error handling for unsuccessful form submission is missing. For instance, if the form validation fails or a required field is missing, try adding Selenium’s in-built exceptions package with from `selenium.common.exceptions import *`. Something like that:

```python
def test_submit_form(setup):
    # All codes are as you shared
    text_box_page.click_submit()

    # Right after clicking the submit button
    try:
        # Outputs
        actual_name, actual_email, actual_current_address, actual_permanent_address = text_box_page.get_output()
    except NoSuchElementException:
        pytest.fail("Form submission may not have been successful, could not find all the expected output fields.")
    # All codes are as you shared
```